core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/OpenAPIGenerator.java

```diff
@@ -3,51 +3,83 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.ObjectWriter;
+import io.metadew.iesi.common.configuration.metadata.repository.MetadataRepositoryConfiguration;
 import io.metadew.iesi.metadata.configuration.component.ComponentConfiguration;
 import io.metadew.iesi.metadata.configuration.connection.ConnectionConfiguration;
+import io.metadew.iesi.metadata.configuration.exception.MetadataAlreadyExistsException;
 import io.metadew.iesi.metadata.definition.component.Component;
 import io.metadew.iesi.metadata.definition.connection.Connection;
+import io.metadew.iesi.metadata.operation.MetadataRepositoryOperation;
+import io.metadew.iesi.metadata.repository.MetadataRepository;
+import lombok.extern.log4j.Log4j2;
 
 import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 
+@Log4j2
 public class OpenAPIGenerator {
     private static OpenAPIGenerator instance;
 
     private OpenAPIGenerator() {}
 
 
-    public static synchronized OpenAPIGenerator getInstance() {
-        if (instance  == null) {
+
+    public synchronized static OpenAPIGenerator getInstance() {
+        if (instance == null) {
             instance = new OpenAPIGenerator();
         }
         return instance;
     }
 
-    public void generate(List<Connection> connections, List<Component> components) {
-
+    public void generate(List<Connection> connections, List<Component> components, String target, boolean load) {
         ObjectMapper mapper = new ObjectMapper();
         ObjectWriter writer = mapper.writer(new DefaultPrettyPrinter());
+
+
         try {
             for (Component component : components) {
-                writer.writeValue(
-                        new File(".." + File.separator + "metadata" + File.separator + "in" + File.separator + "new" +
-                                File.separator +  "component_" + component.getName() + "_v" +
-                                component.getMetadataKey().getVersionNumber() + ".json")
-                        , component);
-                ComponentConfiguration.getInstance().insert(component);
+                saveComponentInDirectory(writer,target, component, load);
+                if (load) saveComponent(component);
             }
             for (Connection connection : connections) {
-                writer.writeValue(
-                        new File(".." + File.separator + "metadata" + File.separator + "in" + File.separator + "new" +
-                                File.separator +  "Connections.json"),
-                        connection);
-                ConnectionConfiguration.getInstance().insert(connection);
+                saveConnectionInDirectory(writer, target,connection, load);
+                if(load) saveConnection(connection);
             }
-
         } catch (IOException e) {
-            e.printStackTrace();
+            log.warn("The target directory doesn't exist, the process is aborted");
```

Comment:
you can't be sure that this is the reason why the process fails. Lets just throw the IOException and let the caller handle it

Reply:
did you have a look at this?